core/sql/generator/GenProbeCache.cpp

```diff
@@ -361,6 +361,38 @@ short ProbeCache::codeGen(Generator *generator)
       // that there is room enough for one row.
     }
 
+   double  memoryLimitPerCpu =
+      ActiveSchemaDB()->getDefaults().getAsLong(EXE_MEMORY_FOR_PROBE_CACHE_IN_MB) * 1024 * 1024;
+   double estimatedMemory;
+   
+   if (numInnerTuples_ > 0) {
+      estimatedMemory = numInnerTuples_ * innerRecLength;
+      if (estimatedMemory > memoryLimitPerCpu) {
+          numInnerTuples_ = memoryLimitPerCpu / innerRecLength;
+          queue_index pUpSize_calc;
+          queue_index pDownSize_calc;                  
+ 
+          pUpSize_calc = numInnerTuples_ * pUpSize / (pUpSize + pDownSize);
+          pDownSize_calc = numInnerTuples_ * pDownSize / (pUpSize + pDownSize);
+          queue_index pq2 = 1;
+          queue_index bits = pUpSize_calc;
+          while (bits && pq2 < pUpSize_calc) {
+              bits = bits  >> 1;
+              pq2 = pq2 << 1;
+          }
+          pUpSize = pq2;
```

Comment:
The code on lines 350-356 makes sure that numInnerTuples_ is greater or equal pUpSize. This while loop seems to me to be doing the opposite, make sure that pUpSize (and pDownSize) below are greater or equal numInnerTuples_. Should this say: pUpSize = pq2 / 2 instead?